var/spack/repos/builtin/packages/snakemake/package.py

```diff
@@ -45,17 +61,25 @@ class Snakemake(PythonPackage):
     depends_on("py-gitpython", type=("build", "run"))
     depends_on("py-psutil", type=("build", "run"))
     depends_on("py-nbformat", type=("build", "run"))
-    depends_on("py-toposort", type=("build", "run"))
+    # See https://github.com/snakemake/snakemake/blob/v7.24.0/setup.cfg#L57
+    depends_on("py-toposort@1.10:", type=("build", "run"), when="@7.24.0:")
+    depends_on("py-toposort", type=("build", "run"), when="@:7.23")
     depends_on("py-connectionpool@0.0.3:", type=("build", "run"))
     depends_on("py-pulp@2:", type=("build", "run"))
     depends_on("py-smart-open@3:", type=("build", "run"))
     depends_on("py-filelock", type=("build", "run"), when="@:6")
     depends_on("py-stopit", type=("build", "run"))
     depends_on("py-tabulate", type=("build", "run"))
     depends_on("py-ratelimiter", type=("build", "run"), when="@:6")
-    depends_on("py-yte@1", type=("build", "run"), when="@7:")
+    # See https://github.com/snakemake/snakemake/blob/v7.28.1/setup.cfg#L59
+    depends_on("py-yte@1.5.1:1", type=("build", "run"), when="@7.28.1:")
+    depends_on("py-yte@1", type=("build", "run"), when="@7:7.28.0")
     depends_on("py-jinja2@3", type=("build", "run"), when="@7:")
     depends_on("py-reretry", type=("build", "run"), when="@7:")
+    # See https://github.com/snakemake/snakemake/blob/v7.28.0/setup.cfg#L60
+    depends_on("py-msrest", type=("build", "run"), when="@7.28.0")
+    # See https://github.com/snakemake/snakemake/blob/v7.29.0/setup.cfg#L48
+    depends_on("py-packaging", type=("build", "run"), when="@7.29.0:")
```

Comment:
Would be much easier to review if all dependencies were in the same order as in `pyproject.toml`. I don't think you need a comment for every dependency, just one for deps found in `# pyproject.toml` and `# setup.cfg`.

Reply:
> I think people should just sort their dependencies alphabetically, but I understand that you want to have some consistent ordering and the rule is to order as in the source material.

The reason why pyproject.toml/setup.py/setup.cfg order is prefered is because @manuelakuhn and I need some way to validate that A) all deps from the file are in the package.py, and B) all deps from the package.py are in the file. If they're in the same order, it's really obvious when they don't match. If not, I have to check both directions to make sure there are no missing and/or extraneous dependencies.

> You already accepted the pull request, do you want me to reorder it? Should I also remove all the comments?

It's optional.

> There are some issues with following the order of a package. For example dependencies which were in previous versions, but not in the current one.

For these, we usually add a `# Historical dependencies` section so as not to clutter the package.py.

> Do you want just two links to both files for easy access?

I prefer to use filenames instead of links since links become out-of-date quickly and filenames rarely change. I actually don't usually document the filenames unless they are different from the usual pyproject.toml/setup.py/setup.cfg, but it can be helpful for people new to Python packaging.

Hope this helps!

Reply:
I have corrected the order and removed comments.